.github/workflows/ci.yml

```diff
@@ -0,0 +1,97 @@
+name: CI
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+
+jobs:
+  test-on-ubuntu:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-18.04, ubuntu-20.04]
+        compiler: [{C: gcc-7, CXX: g++-7}, {C: gcc-11, CXX: g++-11}]
```

Comment:
Worth testing on oldest gcc we support, if that change is easy.  If it's not, fine leaving this at gcc 7 for now

Reply:
Okay. Changed it from gcc-7 to gcc-5 (this required adding additional apt-repositories)